sklearn/gaussian_process/_gpr.py

```diff
@@ -342,14 +341,14 @@ def predict(self, X, return_std=False, return_cov=False):
                 return y_mean
         else:  # Predict based on GP posterior
             K_trans = self.kernel_(X, self.X_train_)
-            y_mean = K_trans.dot(self.alpha_)  # Line 4 (y_mean = f_star)
+            y_mean = K_trans @ self.alpha_  # Line 4 (y_mean = f_star)
 
             # undo normalisation
             y_mean = self._y_train_std * y_mean + self._y_train_mean
 
             if return_cov:
                 v = cho_solve((self.L_, True), K_trans.T)  # Line 5
-                y_cov = self.kernel_(X) - K_trans.dot(v)  # Line 6
+                y_cov = self.kernel_(X) - K_trans @ v  # Line 6
```

Comment:
We try to keep PRs as small as possible. This PR started out as an improvement to docstrings, but now we are touching some code.

Even tho these changes are minimum, I suggest that we revert the changes related to `@` and open another PR.